### PR TITLE
v2.0.0

### LINKED ISSUE #42: Find a way to remove `react-transition-group`
We are shipping a big chunk of JS by using that dependency and that's only used on the nav.
@andrewrubin [suggested](https://github.com/wethegit/corgi/pull/38#discussion_r1230251159) a solution that I think we should make it our default.

### LINKED ISSUE #56: Exclude locales from build
We should be able to exclude a locale from the build, so that when we bootstrap a project with preexisting global locale data, we don't have to move the unused (yet) locale directories out of their homes.

Maybe this is as simple as deleting the localized pages as a `postbuild` action. Open to ideas.

### LINKED ISSUE #59: Nav: clean up HTML elements and ARIA attributes
- We should be using `<ul>` instead of `<menu>`
- The `aria-expanded` attribute should be on the toggle button, not the nav list
  - Let's add `aria-controls` to the main `<nav>` itself

### LINKED ISSUE #66: CLI: Make kebab-casing smarter
As expected, if you scaffold a component called `MyCoolSection`, the folder name will be generated as `my-cool-section`.

We need to make this a bit smarter so that something like `FAQSection` properly converts to `faq-section`, rather than the current behavior of `faqsection`

### LINKED ISSUE #69: CLI: component command: use JSX file extension; create index file
`corgi component MyComponent` should generate the following:

```
my-component/
├── my-component.jsx
├── my-component.module.scss
└── index.js
```

The index file should include a single, named export for the component. 

Optional / up for discussion: Whether to crawl the file system to find nearest existing parent index files, and add the component to those as well. That is probably overkill though

### PR DESCRIPTION
**Pull request by andrewrubin**
# What does this implement or fix?
- Accessibility fixes on the `<Nav>` and `<ReducedMotionButton>` components.
- Refactor YAML compilation to pull from the locale config rather than the filesystem.
- Refactor locale-config to derive locale slugs from the localeMap — ensuring there's not duplicate locale strings being written.
- Move YAML into Webpack bundle.
- Remove React Transition Group dependency.
- Remove `PageTransition` component.
- Switch to NextJS standard NPM script names (i.e. `npm run dev` instead of `npm start`).
- Remove `pageSlug` value from pageProps
- Various QoL improvements
- Various bugfixes

## Testing
To test, `cd` into an empty directory, and run:
```bash
npx @wethegit/corgi@latest project . --template https://github.com/wethegit/corgi-private-templates/tree/dev
```

## Does this close any existing issues?
Closes #42 
Closes #56 
Closes #59 
Closes #66 
Closes #69

### CODE REVIEW

**Comment by andrewrubin** [comment]
Just flagging that this needs to be tested for pages which only exist on certain locales.

**Comment by andrewrubin** [comment]
> Just flagging that this needs to be tested for pages which only exist on certain locales.

Tested this and it works correctly. Removing any locales from a specific subpage will serve the 404 page locally for those locales. Heads-up @marlonmarcello @ste-vg

File: cli/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@wethegit/corgi",
-  "version": "1.0.2",
+  "version": "1.1.0",
```
Comment (liamegan) [inline cli/package.json:3]: I don't think the YAML updates are likely to be at all backwards compatible here, are they? If not, we should look at updating this to a 2.0.0.

**Review by liamegan — COMMENTED** [review]
Looks great, just one tiny semver question for you.

File: cli/package.json (hunk 1 above)
Comment (marlonmarcello) [reply cli/package.json:3]: Good point Liam, I don't think it will be indeed.

Comment (andrewrubin) [reply cli/package.json:3]: Updated PR title to `v2.0.0`.

**Comment by andrewrubin** [comment]
We're adding a lot of updates to this release, just making a note to update the PR description when we're ready for an actual release.

**Review by marlonmarcello — APPROVED** [review]
:shi

**Pull request merged**